src/server/engine_shard_set.cc

```diff
@@ -30,6 +32,17 @@ ABSL_FLAG(bool, cache_mode, false,
           "If true, the backend behaves like a cache, "
           "by evicting entries when getting close to maxmemory limit");
 
+// memory de-fragmentation related flags
```

Comment:
New flags to control the threshold for the task

Reply:
for consistency, I suggest using ratio instead of uint

Reply:
that's not what i meant. use 0.05 instead of 20 - i.e. ratio. you divide by this number and I suggest that you multiply by it.